adafruit_onewire/bus.py

```diff
@@ -0,0 +1,206 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Carter Nelson for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`OneWire Bus`
+====================================================
+
+Provide access to a 1-Wire bus.
+
+* Author(s): Carter Nelson
+"""
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_OneWire.git"
+
+import busio
+from micropython import const
+
+_SEARCH_ROM = const(0xF0)
+_MATCH_ROM = const(0x55)
+_SKIP_ROM = const(0xCC)
+
+class OneWireError(Exception):
+    """A class to represent a 1-Wire exception."""
+    pass
+
+class OneWireAddress(object):
+    """A class to represent a 1-Wire address."""
+
+    def __init__(self, rom):
+        self._rom = rom
+
+    @property
+    def rom(self):
+        """The unique 64 bit ROM code."""
+        return self._rom
+
+    @property
+    def crc(self):
+        """The 8 bit CRC."""
+        return self._rom[7]
+
+    @property
+    def serial_number(self):
+        """The 48 bit serial number."""
+        return self._rom[1:7]
+
+    @property
+    def family_code(self):
+        """The 8 bit family code."""
+        return self._rom[0]
+
+class OneWireBus(object):
+    """A class to represent a 1-Wire bus."""
+
+    def __init__(self, pin):
+        self._ow = busio.OneWire(pin)
+
+    def reset(self, required=False):
+        """
+        Perform a reset and check for presence pulse.
+
+        :param bool required: require presence pulse
+        """
+        reset = self._ow.reset()
+        if required and reset:
+            raise OneWireError
+        return not reset
+
+    def readinto(self, buf, **kwargs):
```

Comment:
**kwargs should be rarely used. Its only really useful when passing kwargs through to a sub-call. Using it otherwise hides incorrect, unused kwargs.

Here simply do `readinto(self, buf, *, start=None, end=None)`. The `*,` tells Python that everything after it must be a kwarg instead of positional.

Reply:
yah, my first time to try using them. got the idea from here:
https://github.com/adafruit/Adafruit_CircuitPython_BusDevice/blob/master/adafruit_bus_device/i2c_device.py#L65